### PR TITLE
Improvements

### LINKED ISSUE #2: support for EU (Europe) region
There's no option for choosing between US en EU regions and all resources are added in the US region.

### PR DESCRIPTION
**Pull request by invidian**
I didn't have time to test it properly yet (that's why WIP), but still would like to share this work, so maybe someone else can test it as well and start using it.

Closes #2

### CODE REVIEW

**Comment by phillbaker** [comment]
Thanks, I won't be able to look at this immediately, but this looks like a
great update.

**Comment by invidian** [comment]
So, I've made some tests and found few bugs, which are fixed now.

I'm not sure yet about parts with context, I just took it from example https://github.com/mailgun/mailgun-go/blob/master/examples/examples.go#L15-L16.

Creating domain and routes works now.

I also added sensitive to smtp_password field, so it does not leak on plan stage.

I also noticed, that `smtp_password` field trigger domain destroy, which is not great, especially, that mailgun-go supports password changing. Perhaps that should be implemented with update-in-place (maybe separated issue, but I don't know if there is a demand for that).

I think it can be reviewed now.

File: resource_mailgun_domain.go [hunk 1]
```diff
@@ -137,19 +144,25 @@ func resourceMailgunDomainCreate(d *schema.ResourceData, meta interface{}) error
...
+	err := client.DeleteDomain(ctx, d.Id())
 	if err != nil {
 		return fmt.Errorf("Error deleting domain: %s", err)
 	}
 
 	// Give the destroy a chance to take effect
 	return resource.Retry(1*time.Minute, func() *resource.RetryError {
-		_, _, _, err = client.GetSingleDomain(d.Id())
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+		defer cancel()
```
Comment (phillbaker) [inline resource_mailgun_domain.go]: 30 seconds is quite a long time for a http request to fail, but it seems interesting to hard code this into the client. I don't think I've seen other timeouts hardcoded into terraform clients.

At a minimum can we extract the `time.seconds*30` into a constant and use it here and below?

**Review by phillbaker — COMMENTED** [review]
Thanks! One small comment before merging.

File: resource_mailgun_domain.go (hunk 1 above)
Comment (invidian) [reply resource_mailgun_domain.go]: It looks like just `context.Background()` should be used, examples:
- https://github.com/terraform-providers/terraform-provider-google/blob/1319ddd86de649e62984da8a31ca5643c4ef1787/google/resource_google_project_services.go#L209
- https://github.com/terraform-providers/terraform-provider-github/blob/6ed5159cce57eee82a11546135579e0b508adf81/github/resource_github_membership.go#L48

So maybe I just change the code to that?

**Comment by phillbaker** [comment]
Sounds good!

**Comment by invidian** [comment]
Updated with plain `context.Background()`.

**Pull request merged**